demos/portal-embed-demo/public/js/ttt/portals-controller.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * Predecessor Controller
+ * Controlling all the portal related features
+ */
+class PortalsController {
+
+    /**
+     * Initiating controller
+     */
+    constructor() {
+        /**
+         * Display mode options
+         */
+        this.RESET = 'RESET';
+        this.EMBED = 'EMBED';
+        this.ACTIVATE = 'ACTIVATE';
+        this.PREDECESSOR_ACTIVATE = 'PREDECESSOR_ACTIVATE';
+
+        // All the UI references
+        this.main = document.querySelector('#main');
+        this.header = document.querySelector('#header');
+        this.detail = document.querySelector('#detail');
+        this.follow = document.querySelector('#follow');
+        this.recommendation = document.querySelector('#recommendation');
+        this.lightbox = document.querySelector('#lightbox');
+        this.embed = document.querySelector('#embed');
+        this.heroImg = document.querySelector('#hero-img');
+        this.audioController = document.querySelector('audio-controller');
+
+        // Add event listeners
+        this._hookEvents();
+    }
+
+    /**
+     * Controll all the UI/animations with given display mode
+     * @param {String} mode 
+     * @param {Object} option 
```

Comment:
If you split the method, you can get proper typing for the arguments that apply to case.

Reply:
Split to `setDefaultUI`, `setEmbedUI`, `setPortalActivatedUI` and `setPredecessorActivateUI`.